plugins/couchbase/src/main/java/org.radargun.service/CouchbaseClientOperations.java

```
@@ -0,0 +1,105 @@
+            return new String((byte[]) object, StandardCharsets.UTF_8);
+         }
+
+         throw new IllegalArgumentException("I don't know hot to decode: " + object + ", class type: " + object.getClass());
+      }
+
+      private byte[] valueAsByteArray(Object object) {
+         if (object instanceof String) {
+            return ((String) object).getBytes();
+         } else if (object instanceof byte[]) {
+            return (byte[]) object;
+         }
+
+         throw new IllegalArgumentException("I don't know hot to decode: " + object + ", class type: " + object.getClass());
+      }
+
+      @Override
+      public V get(K key) {
+         return get(keyAsString(key));
+      }
+
+      private V get(String key) {
+         ByteArrayDocument document = service.bucket.get(key, ByteArrayDocument.class);
+         return (V) document.content();
+      }
+
+      @Override
+      public boolean containsKey(K key) {
+         return service.bucket.exists(keyAsString(key));
+      }
+
+      @Override
+      public void put(K key, V value) {
+         put(keyAsString(key), value);
+      }
+
+      private void put(String key, V value) {
+         ByteArrayDocument doc = ByteArrayDocument.create(key, valueAsByteArray(value));
+         service.bucket.upsert(doc);
+      }
+
+      @Override
+      public V getAndPut(K key, V value) {
+         String keyStr = keyAsString(key);
+         V document = get(keyStr);
+         put(keyStr, value);
+         return document;
+      }
+
+      @Override
+      public boolean remove(K key) {
+         return remove(keyAsString(key));
+      }
+
+      private boolean remove(String key) {
+         try {
+            service.bucket.remove(key);
+            return true;
+         } catch(Exception e) {
+            return false;
+         }
+      }
+
+      @Override
+      public V getAndRemove(K key) {
+         String keyStr = keyAsString(key);
+         V document = get(keyStr);
```

It should be only one operation in the server. In this case, we are calling the server twice. If Couchbase doesn't support returning the previous value when removing the key we need to throw an exception.